microservices/case-service/src/main/java/org/entando/bundle/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,246 @@
+    this.config = config;
+  }
+
+  private String getBucketName() {
+    String name = null;
+
+    try {
+      name = config.getS3().getBucket().getName();
+    } catch (Throwable t) {
+      log.error("Configuration error detected! No bucket name defined", t.getLocalizedMessage());
+    }
+    return name;
+  }
+
+  private String getBucketUrl() {
+    String url = null;
+
+    try {
+      url = config.getS3().getBucket().getUrl();
+    } catch (Throwable t) {
+      log.error("Configuration error detected! No bucket url defined", t.getLocalizedMessage());
+    }
+    return url;
+  }
+
+  private Region getBucketRegion() {
+    Region region = Region.EU_WEST_1;
+
+    try {
+      region = Region.of(config.getS3().getRegion());
+    } catch (Throwable t) {
+      log.error("Configuration error detected! No AWS region defined", t.getLocalizedMessage());
+    }
+    return region;
+  }
+
+  private S3Client getClient() {
+    // Create the S3Client object.
+    return S3Client.builder()
+      .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+      .region(getBucketRegion())
+      .build();
+  }
+
+  @Override
+  public String fileUpload(File file, Map<String, String> metadata) {
+    S3Client s3 = getClient();
+
+    try {
+      PutObjectRequest putOb = PutObjectRequest.builder()
+        .bucket(getBucketName())
+        .key(file.getName())
+        .metadata(metadata)
+        .build();
+      PutObjectResponse response = s3.putObject(putOb, RequestBody.fromBytes(getObjectFile(file.getAbsolutePath())));
+      return response.eTag();
+    } catch (S3Exception t) {
+      log.error(t.awsErrorDetails().errorMessage());
+      t.printStackTrace();
+    }
+    return null;
+  }
+
+  private byte[] getObjectFile(String filePath) {
+    FileInputStream fileInputStream = null;
+    byte[] bytesArray = null;
+
+    try {
+      File file = new File(filePath);
+      bytesArray = new byte[(int) file.length()];
+      fileInputStream = new FileInputStream(file);
+      fileInputStream.read(bytesArray);
+
+    } catch (IOException e) {
+      log.error("error building byte array of file", e);
+    } finally {
+      if (fileInputStream != null) {
+        try {
+          fileInputStream.close();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+    return bytesArray;
+  }
+
+  @Override
+  public List<S3Object> fileList() {
+    S3Client s3 = getClient();
+
+    try {
+      ListObjectsRequest listObjects = ListObjectsRequest
+        .builder()
+        .bucket(getBucketName())
+        .build();
+
+      ListObjectsResponse res = s3.listObjects(listObjects);
+      log.debug("File list successfully downloaded from bucket");
+      return res.contents();
+    } catch (S3Exception t) {
+      log.error("Error getting the file list of a bucket", t.awsErrorDetails().errorMessage());
+    }
+    return null;
+  }
+
+  @Override
+  public File downloadFile(String keyName) {
```

Comment:
Also this operation is probably unneeded as the download will happen just client side by clicking on the file link. Also file download shouldn't pass through a physical file just be streamed to the response.

Reply:
I only did this for the sake of completeness of a minimal file service - it will be removed or refactored as needed, I just coded the simplest case for testing.